clientv3/integration/client_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"bytes"
+	"testing"
+
+	"golang.org/x/net/context"
+
+	"github.com/coreos/etcd/clientv3"
+	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
+	"github.com/coreos/etcd/integration"
+	"github.com/coreos/etcd/lease"
+	"github.com/coreos/etcd/pkg/testutil"
+)
+
+func TestKVPut(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	// TODO: create lease from client side
+	lresp, err := clus.RandClient().Lease.LeaseCreate(
+		context.TODO(),
+		&pb.LeaseCreateRequest{TTL: 1000})
+	if err != nil {
+		t.Fatalf("couldn't lease (%v)", err)
+	}
+	leaseID := lease.LeaseID(lresp.ID)
+
+	tests := []struct {
+		key, val string
+		leaseID  lease.LeaseID
+	}{
+		{"foo", "bar", lease.NoLease},
+		{"foo", "bar", leaseID},
+	}
+
+	for i, tt := range tests {
+		kv := clientv3.NewKV(clus.RandClient())
+
+		if _, err := kv.Put(tt.key, tt.val, tt.leaseID); err != nil {
+			t.Fatalf("#%d: couldn't put (%v)", i, tt.key, err)
+		}
+
+		resp, err := kv.Get(tt.key, 0)
+		if err != nil {
+			t.Fatalf("#%d: couldn't get key (%v)", i, err)
+		}
+		if len(resp.Kvs) != 1 {
+			t.Fatalf("#%d: expected 1 key, got %d", i, len(resp.Kvs))
+		}
+		if !bytes.Equal([]byte(tt.val), resp.Kvs[0].Value) {
+			t.Errorf("#%d: val = %s, want %s", i, tt.val, resp.Kvs[0].Value)
+		}
+		if tt.leaseID != lease.LeaseID(resp.Kvs[0].Lease) {
+			t.Errorf("#%d: val = %d, want %d", i, tt.leaseID, resp.Kvs[0].Lease)
+		}
```

Comment:
@xiang90 Just added lease id check here and `TODO: create lease from client side` above. PTAL.

Thanks!